src/backend/distributed/connection/remote_commands.c

```diff
@@ -152,6 +152,7 @@ NonblockingForgetResults(MultiConnection *connection)
 	while (true)
 	{
 		PGresult *result = NULL;
+		ExecStatusType resultStatus;
 
 		/* just in case there's a lot of results */
 		CHECK_FOR_INTERRUPTS();
```

Comment:
`ClearResultsIfReady()` is called from `StartRemoteTransactionAbort()`. Doesn't checking for interrupts here might cause the transaction abort be aborted and leave database in a bad state? Maybe we should pass a flag to `ClearResultsIfReady` and check for interrupts only if that is true?

Reply:
For example, if query is cancelled here, none of `CurrentCoordinatedTransactionState`, `InProgressTransactions`, or `XactModificationLevel` will be reset. Also some of the clients won't get the `ROLLBACK`, etc. Then the behaviour of next transaction might be unexpected.

Reply:
Agreed, this would probably lead to PANIC and the motivation for keeping it in the comment doesn't seem very strong.